include/deal.II/dofs/dof_tools.h

```diff
@@ -382,11 +400,13 @@ namespace DoFTools
    * This function exists for both non-hp and hp DoFHandlers, to allow for a
    * uniform interface to query this property.
    *
+   * @deprecated Use <code>dh.get_fe_collection().fe_is_primitive()</code>.
+   *
    * @relates hp::DoFHandler
    */
   template <int dim, int spacedim>
   bool
-  fe_is_primitive (const hp::DoFHandler<dim,spacedim> &dh);
+  fe_is_primitive (const hp::DoFHandler<dim,spacedim> &dh) DEAL_II_DEPRECATED;
```

Comment:
Can we really remove `fe_is_primitive`, too? As opposed to the other four functions, there is no `fe_is_primitive()` on `FECollection` (and it is called `is_primitive()` on `FiniteElement`). One can use `dh.get_fe_collection()[0].is_primitive()` as used in the implementation below, though. Can you adjust the documentation?